core/src/jvm_bridge/mod.rs

```diff
@@ -183,7 +183,7 @@ pub fn get_global_jclass(env: &mut JNIEnv, cls: &str) -> JniResult<JClass<'stati
 
     // A hack to make the `JObject` static. This is safe because the global reference is never
     // gc-ed by the JVM before dropping the global reference.
-    let global_obj = unsafe { std::mem::transmute::<_, JObject<'static>>(global.as_obj()) };
+    let global_obj = unsafe { std::mem::transmute::<&JObject<'_>, JObject<'_>>(global.as_obj()) };
```

Comment:
To me the transmute cast from a reference type to a none reference type looks very fishy to me. But I guess this change just make it more explicit that this is what is going on. 

If I change this whole method to 
```
/// Gets a global reference to a Java class.
pub fn get_global_jclass<'a>(env: &mut JNIEnv<'a>, cls: &str) -> JniResult<JClass<'a>> {
    env.find_class(cls)
}
```
the code still builds and seems to test. I guess that is a strong indication that we do not need this hack. 

Created PR removing the method here: https://github.com/apache/datafusion-comet/pull/580